v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/edr-provider.ts

```diff
@@ -421,6 +421,7 @@ async function getProviderConfig(
       const accountOverride: AccountOverride = {
         address,
         balance: BigInt(balance),
+        code: new Uint8Array(), // Empty account code, removing potential delegation code when forking
```

Comment:
For Hardhat 2 we've only done this for "leaked" accounts. That way if the account is a private account by the user we don't unexpectedly remove a delegation.